content/05.dl-complexities.md

```diff
@@ -2,13 +2,13 @@
 
 Correctly training deep neural networks is a non-trivial process.
 There are many different options and potential pitfalls at every stage.
-To get good results you have to expect to train many networks with a range of different parameter and hyperparameter settings.
-Despite improved framework ease-of-use and on-demand cloud computing resources this means DL can be very demanding, often requiring significant infrastructure and patience to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740].
+To get good results, you must expect to train many networks with a range of different parameter and hyperparameter settings.
+Deep learning can be very demanding, often requiring significant computing infrastructure and patience to achieve state-of-the-art performance [@doi:10.1109/JPROC.2017.2761740].
 The experimentation inherent to DL is often noisy (requiring repetition) and represents a significant organizational challenge.
-All code, random seeds, parameters, and results must be carefully corralled using good coding practices (for example, version control, continuous integration etc.) in order to be effective and interpretable.
-This organization is also key to being able to efficiently share your work and to update your model as new data becomes available. 
+All code, random seeds, parameters, and results must be carefully corralled using general good coding practices (for example, version control [@doi:10.1371/journal.pcbi.1004947], continuous integration etc.) in order to be effective and interpretable.
+This organization is also key to being able to efficiently share and reproduce your work [@doi:10.1371/journal.pcbi.1003285; @arxiv:1810.08055] as well as to update your model as new data becomes available.
 
-One specific reproducibility pitfall that is often missed is the default use of non-deterministic algorithms by CUDA/CuDNN backends when training on GPUs.
+One specific reproducibility pitfall that is often missed in deep learning applications is the default use of non-deterministic algorithms by CUDA/CuDNN backends when training models on GPUs.
```

Comment:
technically, these algorithms are also non-deterministic during prediction (after training). It's just not as noticeable. I would rephrase to 

> ... by CUDA/CuDNN backends when ~~training models on~~ [using] GPUs.